project01/dominik/snake.h

```diff
@@ -0,0 +1,38 @@
+#ifndef SNAKE_H
+#define SNAKE_H
+
+#include <vector>
+#include <cinder/app/RendererGl.h>
+
+enum class SnakeMovingDirection
+{
+    left, right, up, down
+};
+
+class Snake
+{
+public:
+    Snake(const cinder::ivec2 &, const int);
+    void Move();
+    void Grow();
+    void ChangeMovingDirection(const SnakeMovingDirection& direction);
+    void IncreaseSpeed() { ++mMovingSpeed; }
+
+    void GetHead(cinder::ivec2& head) const { head = mSnakeBody[0]; }
+    int GetSize() const { return mSnakeBodyPartSize; }
+    void GetBody(std::vector<cinder::ivec2>& body) const { body = mSnakeBody; }
+
+    bool HeadClashedWithBody() const;
+    bool HeadOutsideField() const;
+    bool ObjectClashedWithBody(const cinder::ivec2&) const;
```

Comment:
Why is this needed? Arent the two previous methods enough? Or, are they calling this one?